### PR TITLE
Added Basic externs for PNA: Meter, Counter, Hash, etc

### PR DESCRIPTION
**Pull request by rupesh-chiluka-marvell**
Part of Issue: https://github.com/p4lang/behavioral-model/issues/1245

### CODE REVIEW

**Comment by qobilidop** [comment]
Is there a way we can test the code?

**Comment by antoninbas** [comment]
> Is there a way we can test the code?

I think the priority should be to add some PNA-specific tests (see https://github.com/p4lang/behavioral-model/pull/1255#issuecomment-2161297606).

**Comment by rupesh-chiluka-marvell** [comment]
> > Is there a way we can test the code?
> 
> I think the priority should be to add some PNA-specific tests (see [#1255 (comment)](https://github.com/p4lang/behavioral-model/pull/1255#issuecomment-2161297606)).

Yes. Ok.

File: configure.ac [hunk 1]
```diff
@@ -327,6 +327,7 @@ AC_CONFIG_FILES([Makefile
 		targets/psa_switch/Makefile
 		targets/psa_switch/tests/Makefile
 		targets/pna_nic/Makefile
+        targets/pna_nic/tests/Makefile
```
Comment (qobilidop) [inline configure.ac:330]: Nit: format to align indentation?

File: targets/pna_nic/tests/test_hash.cpp
Comment (qobilidop) [inline targets/pna_nic/tests/test_hash.cpp:1]: It does feel there could be some common code factored out to test both `psa_switch` and `pna_nic`'s `hash` and `internet_checksum` blocks. But that could probably be left as future work.

**Review by qobilidop — APPROVED** [review]
Thanks for adding the tests! Now it looks good from my side.

File: configure.ac (hunk 1 above)
Comment (rupesh-chiluka-marvell) [reply configure.ac:330]: Aligned it.
tabs are used for indentation in configuration files. Everywhere else, spaces are used.

File: targets/pna_nic/tests/test_hash.cpp
Comment (rupesh-chiluka-marvell) [reply targets/pna_nic/tests/test_hash.cpp:1]: Yes. I was thinking about how to do it. Based on future tests, we can factor out common code.

**Comment by antoninbas** [comment]
@rupesh-chiluka-marvell please break up this PR. Everything related to the `pna-demo-L2-one-table.json` test should be in its own PR, which I'd like to review & merge first. Unless I am missing something, this basic end-to-end test of PNA has no dependencies on the externs you want to add as par of this PR.

**Comment by rupesh-chiluka-marvell** [comment]
> @rupesh-chiluka-marvell please break up this PR. Everything related to the `pna-demo-L2-one-table.json` test should be in its own PR, which I'd like to review & merge first. Unless I am missing something, this basic end-to-end test of PNA has no dependencies on the externs you want to add as par of this PR.

Yes. Ok.

**Comment by rupesh-chiluka-marvell** [comment]
> @rupesh-chiluka-marvell please break up this PR. Everything related to the `pna-demo-L2-one-table.json` test should be in its own PR, which I'd like to review & merge first. Unless I am missing something, this basic end-to-end test of PNA has no dependencies on the externs you want to add as par of this PR.

Hi @antoninbas, I created a new PR for the basic test of PNA here: https://github.com/p4lang/behavioral-model/pull/1262. Please review it.

**Pull request closed without merging**